### PR TITLE
Add thruster deadband

### PR DESCRIPTION
**Pull request by gabrielfpacheco**
This PR adds to the thruster simulation a real behavior, the thruster deadband, i.e. a specific region of low speed in which the motors are not activated and, consequently, no thrust is generated. The thrust-speed relationship is modeled as:

<a href="https://www.codecogs.com/eqnedit.php?latex=\tau&space;=&space;C_t&space;\;&space;\omega&space;\left&space;|&space;\omega&space;\right&space;|" target="_blank"></a>

where, 

<a href="https://www.codecogs.com/eqnedit.php?latex=C_t&space;=&space;2.358&space;\times&space;10^{-3}&space;N.s^2/rad^2" target="_blank"></a>

for Kraken T100V048 thrusters. This coeeficient has been identified using the  least-squares with data taken from: http://krakenpower.de/wp-content/uploads/2018/01/Datasheet_RDT_T100_KPG-2.pdf 

Depends on
----------------
* [ ] https://github.com/ThirteenLtda/bundles-squidbot/pull/83
* [ ] https://github.com/ThirteenLtda/thirteen.squidbot-package_set/pull/39
* [ ] https://github.com/ThirteenLtda/simulation-orogen-rock_gazebo/pull/2

How to Test
----------------
Execute the following commands after setting the correct branches. As the modifications were in the simulation side, the test need to be executed running the complete simulation:

* Open a terminal and start `squid plan sketch`.

```bash
# terminal 1
acd bundles/squidbot
squid plan sketch
```

* Start `squidbot` system. You can use `syskit ide` for that.

```bash
# terminal 2
syskit ide -rgazebo
```
* For testing, run the `move_to_position` action. You can use `syskit ide` for that.

```
move_to_position!(
  target_pose: Hash[position: Eigen::Vector3.new(2, 3, 5), yaw: 0]
)
```

* Check if the action was stopped after the vehicle reached the position. You can test other positions.

Expected Result
----------------
Once within deadband (|effort| < 1.6 approximately), one should obtain a different effort sent by `squid_accelerator_controller` than the one generated by `gazebo_thruster` as shown in the figure below:

image

### CODE REVIEW

File: src/GazeboThrusterTypes.hpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#include <string>
+
+namespace gazebo_thruster
+{
+
+    struct Thruster
+    {
+        std::string name; /*! Thruster name in the SDF file.*/
+        double effort; /*! Current effort in [N] generated by the thruster.*/
+        double minThrust; /*! Minimum effort in [N] generated by the thruster.*/
```
Comment (gustavoneves12) [inline src/GazeboThrusterTypes.hpp]: Please, use the same pattern for naming the variables. Usually, as in google style code, I usually set the variable names of structures using `snake_case` style (See here: https://google.github.io/styleguide/cppguide.html#Variable_Names).

**Comment by gustavoneves12** [comment]
I tested here, it worked as expected.

File: src/GazeboThrusterTypes.hpp (hunk 1 above)
Comment (gabrielfpacheco) [reply src/GazeboThrusterTypes.hpp]: Please check https://github.com/ThirteenLtda/simulation-gazebo_thruster/pull/3/commits/a0558830ea199a18c6c5573439edb2bf19180bda

File: src/GazeboThruster.cpp [hunk 2]
```diff
@@ -68,64 +50,58 @@ std::vector<gazebo_thruster::GazeboThruster::Thruster> GazeboThruster::loadThrus
...
         thrusterElement = thrusterElement->GetNextElement("thruster");
     }
-
+    
     if (thrusters.empty())
     {
-        string msg = "GazeboThruster: sdf model loads thruster plugin but has no thruster defined.\n";
-        msg += "GazeboThruster: please name the links you want to export as thrusters inside the <plugin> tag: \n";
+        string msg = "GazeboThruster: sdf model loads thruster plugin";
+        msg +=  "but has no thruster defined.\n";
```
[the remark this answers is not in the document]
Comment (gabrielfpacheco) [reply src/GazeboThruster.cpp]: Please check https://github.com/ThirteenLtda/simulation-gazebo_thruster/pull/3/commits/87ea27240fc0463c7bcd9830901d4703657ed80e

File: src/GazeboThruster.cpp [hunk 3]
```diff
@@ -68,64 +50,58 @@ std::vector<gazebo_thruster::GazeboThruster::Thruster> GazeboThruster::loadThrus
...
+        msg += "but it cannot be found in the SDF object.\n";
+        msg += "GazeboThruster: expected the thruster plugin filename to be";
+        msg += "libgazebo_thruster.so\n";
         gzthrow(msg);
     }
 
-    gzmsg << "GazeboThruster: found plugin (filename): " << pluginElement->Get<string>("filename") << endl;
-    gzmsg << "GazeboThruster: found plugin (name): " << pluginElement->Get<string>("name") << endl;
+    gzmsg << "GazeboThruster: found plugin (filename): " 
+            << pluginElement->Get<string>("filename") << endl;
```
[the remark this answers is not in the document]
Comment (gabrielfpacheco) [reply src/GazeboThruster.cpp]: Please check https://github.com/ThirteenLtda/simulation-gazebo_thruster/pull/3/commits/87ea27240fc0463c7bcd9830901d4703657ed80e

File: src/GazeboThruster.cpp [hunk 4]
```diff
@@ -21,7 +21,7 @@ void GazeboThruster::Load(physics::ModelPtr _model,sdf::ElementPtr _sdf)
     model = _model;
     gzmsg << "GazeboThruster: loading thrusters from model: " << model->GetName() << endl;
 
-    std::vector<Thruster> thrusters = loadThrusters();
```
[the remark this answers is not in the document]
Comment (gabrielfpacheco) [reply src/GazeboThruster.cpp]: Please check https://github.com/ThirteenLtda/simulation-gazebo_thruster/pull/3/commits/87ea27240fc0463c7bcd9830901d4703657ed80e

**Comment by geovanemimoso** [comment]
Tested. Works fine, nice job @gabrielfpacheco ;)

**Pull request merged**